src/main/java/com/faforever/client/game/GameDetailController.java

```diff
@@ -167,6 +146,22 @@ public void resetListeners() {
     teamsInvalidationListener = observable -> createTeams();
     numPlayersInvalidationListener = observable -> onNumPlayersChanged();
     gamePropertiesInvalidationListener = observable -> onGamePropertyChanged();
+    startTimeInvalidationListener = observable -> onStartTimeChanged();
+  }
+
+  private void onStartTimeChanged() {
+    Optional.ofNullable(game.get())
+        .filter(gameBean -> gameBean.getStatus() == GameStatus.PLAYING)
+        .map(GameBean::getStartTime)
+        .ifPresentOrElse(startTime -> JavaFxUtil.runLater(() -> {
+              watchButtonController.setGame(game.get());
+              joinButton.setVisible(false);
+              watchButton.setVisible(true);
+            }),
+            () -> JavaFxUtil.runLater(() -> {
+              joinButton.setVisible(game.get() != null && game.get().getStatus() == GameStatus.OPEN);
+              watchButton.setVisible(false);
+            }));
```

Comment:
There is no real reason to put this inside an optional. I would just do a null check at the beginning and just return if null rather than wrapping in an optional as this just creates an object for little benefit.